Peer Review

baasbank opened the pull request:
## What does this PR do?
This PR enables a peer to review the Articools project.

baasbank committed:
define routes

baasbank committed:
update .eslint config

baasbank committed:
update gitignore

baasbank committed:
integrate travis CI

baasbank committed:
add jest for tests

baasbank committed:
add test script

baasbank committed:
add dummy test

baasbank committed:
add travis passing badge

baasbank committed:
update README

baasbank committed:
create the .hound.yml file

baasbank committed:
remove utils folder

baasbank committed:
add react-google-login

baasbank committed:
import login component

baasbank committed:
get user profile details from local storage

baasbank committed:
add login component

baasbank committed:
add coveralls badge

baasbank committed:
integrate coveralls

baasbank committed:
update this component

baasbank committed:
update this component

baasbank committed:
update this componet

baasbank committed:
update README

baasbank committed:
add coveralls script

baasbank committed:
update travis file

baasbank committed:
add .babelrc file

baasbank committed:
modify rules

baasbank committed:
update gitignore

baasbank committed:
update travis to run coverage

baasbank committed:
add jest script

baasbank committed:
update webpack config

baasbank committed:
update package.json

baasbank committed:
update webpack config file

baasbank committed:
rewrite test

baasbank committed:
create scss rules

baasbank committed:
update file

baasbank committed:
update file

baasbank committed:
create footer component

baasbank committed:
refactor component

baasbank committed:
update file

baasbank committed:
update login component

baasbank committed:
create navbar component

baasbank committed:
create topbar component

baasbank committed:
test actions

baasbank committed:
update files

baasbank committed:
configure the server

baasbank committed:
set up server

baasbank committed:
add dependencies

baasbank committed:
update package.json

baasbank committed:
update package.json

baasbank committed:
update package.json

baasbank committed:
update login route

baasbank committed:
refactoring

baasbank committed:
refactor Sources.js

baasbank committed:
refactoring

baasbank committed:
refactoring

baasbank committed:
reconfigure the app

baasbank committed:
add JSDOC comments

baasbank committed:
creates the page not found component

baasbank committed:
add JSDOC comments

baasbank committed:
add JSDOC comments

baasbank committed:
Update README.md

baasbank committed:
Update README.md

baasbank committed:
Update README.md

baasbank committed:
write actions test

baasbank committed:
Merge branch 'testing' of https://github.com/baasbank/headlines into testing

update README

baasbank committed:
update node version in travis.yml

baasbank committed:
update travis.yml

baasbank committed:
update node version in travis.yml

baasbank committed:
creates .coveralls.yml file

baasbank committed:
Update README.md

baasbank committed:
Update README.md

baasbank committed:
update README

baasbank committed:
install coveralls

baasbank committed:
add codeclimate badge

baasbank committed:
change text color on login page

baasbank committed:
test articles store

baasbank committed:
Merge pull request #1 from baasbank/testing

Testing

baasbank committed:
change badges branch to master

raphaelakpan commented:
On the hosted application, after I have successfully logged in with my gmail and go to the homepage by clicking on `Articools`, I still see `Login with Google` displayed on the homepage despite the fact that I am already logged in. Secondly, I only see the `Logout` button, but nothing shows my name or info for me to be sure I am the one logged in. I think you should fix those two things.

raphaelakpan commented:
Also, when I click on a news source, say `Al-Jazeera`, it displays the news, but does not show which news source I am currently accessing, and whether I am accessing `Latest News`  or  `Top news`. You don't want to leave the user is a state of confusion. Always try as much as possible to ensure the user is never lost or confused on your application.

raphaelakpan commented on .babelrc:
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015", "react", "stage-2"]
+}
```
Always put a new line at the end of every file. Apply this to every other file in the project

baasbank replied:
Fixed.

raphaelakpan commented on README.md:
```diff
@@ -1,2 +1,55 @@
+## Limitations
+
+* Users can only sign in with their google accounts.
+* Articles open only on their source website.
```
You can add the `Author` section to give inform about the author of this project.

baasbank replied:
Fixed.

raphaelakpan commented on sass/main.scss:
```diff
@@ -0,0 +1,177 @@
+//   margin-bottom: 10px;
+// }
+
+#topbarh1{
```
This element id name is sort of fuzzy. You should make your variables clear and unambiguous.

baasbank replied:
Fixed.

raphaelakpan commented on sass/main.scss:
```diff
@@ -0,0 +1,177 @@
+  padding: 0;
+}
+
+#sortbytext li{
```
Remove this style if you are no longer using it.

baasbank replied:
Fixed.

raphaelakpan commented on sass/main.scss:
```diff
@@ -0,0 +1,177 @@
+  padding: 0px;
+}
+
+#topbar{
```
Is it a good code practise to put a space between the named object and curly bracket.

baasbank replied:
Fixed.

raphaelakpan commented on src/app/components/PageNotFound.jsx:
```diff
@@ -0,0 +1,16 @@
+  <div>
+    <h3 style={{ textAlign: 'center' }}>
+      404 Error... Testing me huh????
+    </h3>
```
You can add a link or button to take the user back to the homepage or news sources page.

baasbank replied:
Done.

baasbank committed:
refactor test

baasbank committed:
implement feedback

baasbank committed:
gitignore .env file

baasbank committed:
refactor code

baasbank committed:
fix bug

baasbank committed:
handle axios error

baasbank committed:
implement feedback

baasbank committed:
refactor code

baasbank committed:
refactor code

baasbank committed:
update procfile

baasbank committed:
display user name

baasbank committed:
parse user name

baasbank committed:
Merge pull request #3 from baasbank/feedback

Feedback

baasbank committed:
remove unnecessary comments

baasbank committed:
remove unnecessary comments

baasbank committed:
feature(feedback): remove .env from github
rename General.jsx to App.jsx
deprecate react and react-dom version to remove errors
remove console.log from server.js

baasbank committed:
Merge branch 'master' of https://github.com/baasbank/headlines into feedback

baasbank committed:
adds articles test
adds login component test
adds LoginController component test
add PageNotFound test
add Sources component test
modified the test description in ArticlesStore test from object to array
modify the test description in SourcesStore test from object to array
rewrite Js
refactor Articles component
rewrite JS doc comment in Login component
refactor Sources component

baasbank committed:
Merge pull request #4 from baasbank/feedback

Feedback

baasbank committed:
rewrite the test descriptions for actions
rewrite the test description and test that fetchArticles method is called for Articles
rewrite the test description for LoginController
rewrite the test description for PageNotFound
rewrite the test description and test that fetchSources method is called for Sources
rewrite the test description for ArticlesStore
rewrite the test description for SourcesStore

baasbank committed:
fix misplaced semi colon

baasbank committed:
refactor code

baasbank committed:
refactor code

baasbank committed:
Merge pull request #5 from baasbank/feedback

Feedback Implementation
